contrib/babelfishpg_tsql/src/linked_servers.c

```diff
@@ -102,12 +145,38 @@ linked_server_err_handler(LinkedServerProcess lsproc, int severity, int db_error
 {
 	StringInfoData buf;
 
+	char* err_msg = NULL;
+	char* str =  NULL;
+	char* pos = NULL;
+
 	initStringInfo(&buf);
 
+	if (db_err_str)
+	{
+		/*
+		 * We remove "Adaptive" from error message since we are only
+		 * supporting remote servers that use T-SQL and communicate over TDS
+		 */
+		err_msg = remove_substr(pnstrdup(db_err_str, strlen(db_err_str) + 1), "Adaptive ");
+		str = err_msg;
+
+		/* We convert the 'S' in "Server" to lowercase */
+		while((pos = strstr(str, "Server")) != NULL)
+		{
+			*pos = 's';
+
+			/* skip length of "Server" */
+			str = pos + 6;
+
+			if (!str)
```

Comment:
this is incorrect. You should either check for string length or NULL character. str pointer will never be NULL